src/pages/DetailGrade/index.js

```diff
@@ -112,10 +168,107 @@ const DetailGrades = () => {
       ),
     },
   ])
-  let { id } = useParams()
+  const handleCloseDialog = () => {
+    setDialogMsg(null)
+  }
+  const handleMarkGrades = async (event) => {
+    event.preventDefault()
+    const gradeId = event.target.id
+    const grade = listGradeStudent.filter((g) => g.id == gradeId)
+    if (grade[0].finalized === true) {
+      setDialogMsg('Grade already marked finalized')
+      return
+    }
+    try {
+      const response = await axiosClient.post(
+        `/api/classrooms/${id}/grades/${gradeId}/finalized`
+      )
+      setListGradeStudent((prev) =>
+        prev.map((row) =>
+          row.id === grade[0].id ? { ...row, finalized: true } : row
+        )
+      )
```

Comment:
I think you can use array.find for this case
```suggestion
    const grade = listGradeStudent.filter((g) => g.id == gradeId)
    if (grade.finalized === true) {
      setDialogMsg('Grade already marked finalized')
      return
    }
    try {
      const response = await axiosClient.post(
        `/api/classrooms/${id}/grades/${gradeId}/finalized`
      )
      setListGradeStudent((prev) =>
        prev.map((row) =>
          row.id === grade.id ? { ...row, finalized: true } : row
        )
      )
```